tests/runtest.sh

```diff
@@ -418,8 +435,21 @@ function run_test {
     local outputFileName=$(basename "$outputFilePath")
 
     # Convert DOS line endings to Unix if needed
-    sed -i 's/\r$//' "$scriptFileName"
+    local OSName=$(uname -s)
+    case $OSName in
+        Linux)
+            sed -i 's/\r$//' "$scriptFileName"
+            ;;
 
+        Darwin)
+            perl -pi -e 's/\r\n|\n|\r/\n/g' "$scriptFileName"
+            ;;
+        *)
+            echo "Unsupported OS $OSName detected, configuring as if for Linux"
+            sed -i 's/\r$//' "$scriptFileName"
+            ;;
+    esac
```

Comment:
As with the above, it would be nice if this were unified. I haven't tried them myself, but there seem to be some ways to get the old `sed` on OS X to work properly with carriage returns (for example, by using `$(printf '\r')` rather than just `\r`). Alternatively, if the perl way works on both OS's, then that's fine too.

Please give that a quick shot if you can. If neither of those is a good option, I'm okay with this as is.